ctapipe/io/simteleventsource.py

```diff
@@ -339,19 +349,11 @@ def prepare_subarray_info(self, telescope_descriptions, header):
             else:
                 focal_length = nominal_focal_length
 
-            optics = OpticsDescription(
-                name=telescope.name,
-                num_mirrors=telescope.n_mirrors,
-                equivalent_focal_length=focal_length,
-                mirror_area=mirror_area,
-                num_mirror_tiles=cam_settings["n_mirrors"],
-            )
-
             camera = build_camera(
```

Comment:
One (perhaps less common) use case that might not work here, and also didn't before, is if the user changes the focal length choice after construction.  In that case, the SubarrayDescription should be re-built. :
```py3
with EventSource(filename) as source:
    source.focal_length_choice='nominal'
    source.subarray.tel[1].camera.geometry.frame # still uses effective 
```
We could add a callback to that trait, and either just raise an exception if they try to change it, or else re-generate the CameraFrames.

Reply:
added #1997

Reply:
I think at least `allowed_tels` is also not changeable after construction.